File: backend/Services/Parsers/HertlParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class HertlParser : BasicScoreResultParser
    {
        private ILogger<HertlParser> _logger;

        public HertlParser(ILogger<HertlParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string GameName => "Hertl";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($@"HERTL \(Game #\d+\) - (?<{ScoreGroup}>[\d|X]) / \d");
        protected override string? ExtraContent => Url;
        public override string Url => "https://www.hertl.app";
    }
}

File: lambdas/src/parsers/hertl-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class HertlParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Hertl';
  readonly helpText = null;
  readonly url = 'https://www.hertl.app';
  protected readonly extraContent = this.url;
  
  // Matches "HERTL (Game #813) - X / 8" or "HERTL (Game #690) - 5 / 8"
  // Note: C# uses [\d|X] which incorrectly includes literal pipe - fixed here to [\dX]
  protected readonly parser = new RegExp(
    `HERTL \\(Game #\\d+\\) - (?<${BasicScoreResultParser.SCORE_GROUP}>[\\dX]) / \\d`
  );
}
